corehq/apps/domain/views.py

```diff
@@ -2071,6 +2080,57 @@ def post(self, request, *args, **kwargs):
         return self.get(request, *args, **kwargs)
 
 
+class CaseSearchConfigView(BaseAdminProjectSettingsView):
+    urlname = 'case_search_config'
+    page_title = ugettext_lazy('Case Search')
+    template_name = 'domain/admin/case_search.html'
+
+    @method_decorator(domain_admin_required)
+    @use_bootstrap3
+    def dispatch(self, request, *args, **kwargs):
+        return super(CaseSearchConfigView, self).dispatch(request, *args, **kwargs)
+
+    def post(self, request, *args, **kwargs):
+
+        def unpack_fuzzies(query_dict):
+            """
+            Builds an integer-keyed dictionary from POST request data, and returns a list of dictionaries that can
+            be wrapped by CaseSearchConfigJSON
+            """
+            pattern = re.compile(r'^config\[fuzzy_properties]\[(?P<index>\d+)]\[(?P<attr>\w+)]$')
+            fuzzy_dict = defaultdict(dict)
+            for key, value in query_dict.items():
+                match = pattern.match(key)
+                if match:  # non-fuzzy-properties won't match, i.e. "^enable$"
+                    fuzzy_dict[int(match.group('index'))][match.group('attr')] = value
+            if not fuzzy_dict:
+                return []
+            return [fuzzy_dict[i] for i in range(max(fuzzy_dict.keys()) + 1) if fuzzy_dict[i]]
+
+        try:
+            case_search_config = CaseSearchConfig.objects.get(pk=self.domain)
```

Comment:
I think you could use `CaseSearchConfig.objects.get_or_create` here. or does that save it immediately, which is why you did it this way?